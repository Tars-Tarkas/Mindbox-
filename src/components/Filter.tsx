import React, { useContext, useState, MouseEvent } from "react";
import { Button, Flex } from "antd";
import { ToDoContext } from "../context/TodoContext";
import type { ToDoContextType } from "../types/todo";

const buttonObj = [
  { id: "all", text: "All" },
  { id: "active", text: "Active" },
  { id: "completed", text: "Completed" },
];

export default function Filter() {
  const { setFilter, todos } = useContext(ToDoContext) as ToDoContextType;
  const [active, setActiv] = useState("");

  const handleFilter = (e: MouseEvent<HTMLElement>) => {
    const id = e.currentTarget.id;
    setActiv(id);
    if (id === "all") {
      setFilter(todos);
    }
    if (id === "active") {
      setFilter(
        [...todos].filter((todo) => {
          if (todo.completed === false) {
            return todo;
          }
        })
      );
    }
    if (id === "completed") {
      setFilter(
        [...todos].filter((todo) => {
          if (todo.completed === true) {
            return todo;
          }
        })
      );
    }
  };

  return (
    <Flex justify="space-between" style={{ gap: "10px" }}>
      {buttonObj.map((btn, index) => {
        return (
          <Button
            disabled={todos.length === 0}
            color="default"
            key={index}
            name={btn.id}
            id={btn.id}
            variant={active !== btn.id ? "filled" : "outlined"}
            onClick={handleFilter}
          >
            {btn.text}
          </Button>
        );
      })}
    </Flex>
  );
}
