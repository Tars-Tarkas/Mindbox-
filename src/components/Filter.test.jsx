import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import TodoProvider from "../context/TodoContext";
import Filter from "./Filter";

describe("Filter component", () => {
  it("Filter render", () => {
    render(
      <TodoProvider>
        <Filter />
      </TodoProvider>
    );
    expect(screen.getByRole("button", { name: "All" })).toHaveAttribute(
      "disabled"
    );
    expect(screen.getByRole("button", { name: "Active" })).toHaveAttribute(
      "disabled"
    );
    expect(screen.getByRole("button", { name: "Completed" })).toHaveAttribute(
      "disabled"
    );
  });
  it("Filter snapshot", () => {
    const filter = render(
      <TodoProvider>
        <Filter />
      </TodoProvider>
    );
    expect(filter).toMatchSnapshot();
  });
});
