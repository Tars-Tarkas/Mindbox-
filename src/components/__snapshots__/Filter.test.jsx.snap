// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component Filter snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-flex css-dev-only-do-not-override-1hpnbz2 ant-flex-justify-space-between"
        style="gap: 10px;"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-1hpnbz2 ant-btn-default ant-btn-color-default ant-btn-variant-filled"
          disabled=""
          id="all"
          name="all"
          type="button"
        >
          <span>
            All
          </span>
        </button>
        <button
          class="ant-btn css-dev-only-do-not-override-1hpnbz2 ant-btn-default ant-btn-color-default ant-btn-variant-filled"
          disabled=""
          id="active"
          name="active"
          type="button"
        >
          <span>
            Active
          </span>
        </button>
        <button
          class="ant-btn css-dev-only-do-not-override-1hpnbz2 ant-btn-default ant-btn-color-default ant-btn-variant-filled"
          disabled=""
          id="completed"
          name="completed"
          type="button"
        >
          <span>
            Completed
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-flex css-dev-only-do-not-override-1hpnbz2 ant-flex-justify-space-between"
      style="gap: 10px;"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-1hpnbz2 ant-btn-default ant-btn-color-default ant-btn-variant-filled"
        disabled=""
        id="all"
        name="all"
        type="button"
      >
        <span>
          All
        </span>
      </button>
      <button
        class="ant-btn css-dev-only-do-not-override-1hpnbz2 ant-btn-default ant-btn-color-default ant-btn-variant-filled"
        disabled=""
        id="active"
        name="active"
        type="button"
      >
        <span>
          Active
        </span>
      </button>
      <button
        class="ant-btn css-dev-only-do-not-override-1hpnbz2 ant-btn-default ant-btn-color-default ant-btn-variant-filled"
        disabled=""
        id="completed"
        name="completed"
        type="button"
      >
        <span>
          Completed
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
